src/lib.rs

```diff
@@ -61,25 +68,112 @@ use nrf52832_hal::{
         CorePeripherals,
         Interrupt,
         Peripherals,
+        SPIM2,
+        TWIM1,
     },
     spim,
     twim,
+    uarte::{
+        Parity as UartParity,
+        Baudrate as UartBaudrate,
+    },
     Timer,
     Spim,
     Twim,
 };
 
 #[cfg(feature = "dev")]
 use nrf52832_hal::{
-    gpio::Pin,
+    gpio::{
+        p0::{
+            P0_05,
+            P0_11,
+        },
+        Pin,
+    },
+    nrf52832_pac::{
+        UARTE0,
+    },
     uarte::{
         self,
         Uarte,
-        Parity as UartParity,
-        Baudrate as UartBaudrate,
     },
 };
 
+/// Create a new instance the serial port connected to the debugger,
+/// mapped to the host via USB-Serial
+#[cfg(feature = "dev")]
+pub fn new_usb_uarte<TX, RX>(
+    uart0: UARTE0,
+    txd_pin: P0_05<TX>,
+    rxd_pin: P0_11<RX>,
+    config: UsbUarteConfig
+) -> Uarte<nrf52::UARTE0> {
+    uart0.constrain(uarte::Pins {
+            txd: txd_pin.into_push_pull_output(Level::High).degrade(),
+            rxd: rxd_pin.into_floating_input().degrade(),
+            cts: None,
+            rts: None,
+        },
+        config.parity,
+        config.baudrate
+    )
+}
+
+/// Create a new instance of the DW1000 radio
+pub fn new_dw1000<SCK, MOSI, MISO, CS>(
+    spim: SPIM2,
+    sck: P0_16<SCK>,
+    mosi: P0_20<MOSI>,
+    miso: P0_18<MISO>,
+    cs: P0_17<CS>,
+) -> DW1000<
+        Spim<nrf52::SPIM2>,
+        p0::P0_17<Output<PushPull>>,
+        dw1000::Uninitialized>
+{
+    let spim = spim.constrain(spim::Pins {
+            sck : sck.into_push_pull_output(Level::Low).degrade(),
+            mosi: Some(mosi.into_push_pull_output(Level::Low).degrade()),
+            miso: Some(miso.into_floating_input().degrade()),
+        });
+
+    DW1000::new(spim, cs.into_push_pull_output(Level::High))
+}
+
+/// Create a new instance of the TWIM bus used for the accelerometer
+pub fn new_acc_twim<SCL, SDA>(
+    twim: TWIM1,
+    scl: P0_28<SCL>,
+    sda: P0_29<SDA>,
+) -> Twim<nrf52::TWIM1> {
+    twim.constrain(
+        twim::Pins {
+            scl: scl.into_floating_input().degrade(),
+            sda: sda.into_floating_input().degrade(),
+        },
+        twim::Frequency::K250,
+    )
+}
+
+
+/// Configuration parameters for the UART connected via the debugger
+pub struct UsbUarteConfig {
+    /// Parity setting
+    pub parity: UartParity,
+
+    /// Baudrate setting
+    pub baudrate: UartBaudrate,
+}
+
+impl Default for UsbUarteConfig {
+    fn default() -> UsbUarteConfig {
+        UsbUarteConfig {
+            parity: UartParity::EXCLUDED,
+            baudrate: UartBaudrate::BAUD115200,
```

Comment:
Oh, this is a change from the previous behavior. I can revert this to the (unstable) 460800 baud rate from before, but 115200 is probably a better default.

Reply:
Well, if the higher rate is unstable, this is basically a bug fix. On top of that, it's probably unlikely this change is breaking any real code.

I'm fine with releasing this as a patch version.